misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JBallerinaDebugServer.java

```diff
@@ -282,6 +283,13 @@ ClientConfigHolder getClientConfigHolder() {
             if (loadedThreadFrames.containsKey(activeThread.uniqueID())) {
                 stackTraceResponse.setStackFrames(loadedThreadFrames.get(activeThread.uniqueID()));
             } else {
+                // if the active thread's top stack frame name is `$init$`, then send a `STEP_INTO` request.
+                // This is to avoid `$init$` stack frame information represented on the top of the call stack,
+                // as this stack frame does not give any useful information to the user.
+                if (activeThread.frames().get(0).location().method().name().equals(METHOD_INIT)) {
```

Comment:
Don't we need to handle NPEs and other exceptions when directly accessing these stack frame information?

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/31506/commits/c351d222a78a9ae0aad3971a59c71a74e7c42dfc